internal/compliance/snapshot_poller/pollers/aws/s3_bucket.go

```diff
@@ -355,6 +364,15 @@ func PollS3Buckets(pollerInput *awsmodels.ResourcePollerInput) ([]apimodels.AddR
 			zap.L().Debug("nil bucket returned by S3 list buckets")
 			continue
 		}
+		// Check if ResourceID matches the integration's regex filter
+		if ignore, err := pollerInput.ShouldIgnoreResource(*bucket.Name); ignore || err != nil {
+			if err != nil {
+				return nil, nil, err
+			}
+			if ignore {
+				continue
+			}
+		}
```

Comment:
```suggestion
		if ignore, err := pollerInput.ShouldIgnoreResource(*bucket.Name); ignore || err != nil {
			if err != nil {
				return nil, nil, err
			}
			continue
		}
```

I think we can drop the `if ignore`, we know either `ignore` or `err != nil` is true, and since we didn't enter the above block we know it must be `ignore`.